qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -329,23 +369,54 @@ def reduce(self, atol: Optional[float] = None, rtol: Optional[float] = None) ->
             rtol = self.rtol
 
         label_list, indexes = np.unique(self._labels, return_inverse=True, axis=0)
-        coeff_list = [0] * len(self._coeffs)
+        coeff_list = np.zeros(len(self._coeffs), dtype=np.complex128)
         for i, val in zip(indexes, self._coeffs):
             coeff_list[i] += val
         non_zero = [
-            i
-            for i, v in enumerate(coeff_list)
-            if not np.isclose(v, 0, atol=atol, rtol=rtol)
+            i for i, v in enumerate(coeff_list) if not np.isclose(v, 0, atol=atol, rtol=rtol)
         ]
         if not non_zero:
-            return FermionicOp(("I" * self.register_length, 0))
-        new_labels = label_list[non_zero].tolist()
-        new_coeffs = np.array(coeff_list)[non_zero].tolist()
-        return FermionicOp(list(zip(new_labels, new_coeffs)))
+            return FermionicOp([("I" * self.register_length, 0)])
+        return FermionicOp(list(zip(label_list[non_zero].tolist(), coeff_list[non_zero])))
 
     def __len__(self):
         return len(self._labels)
 
     @staticmethod
     def _validate_label(label: str) -> bool:
+        if not label:
+            return False
         return set(label).issubset({"I", "+", "-", "N", "E"})
+
+    def _from_sparse_label(self, list_label):
+        num_terms = len(list_label)
+        parsed_data = []
+        max_index = 0
+        for term, label in enumerate(list_label):
+            prev_index = -1
+            list_label = label.split()
+            for single_label in list_label:
+                op_label, index_str = single_label.split("_", 1)
+
+                index = int(index_str)
+                if 0 <= prev_index <= index:
+                    raise ValueError("Indices of labels must be in descending order.")
+
+                max_index = max(max_index, index)
+
+                if op_label in {"+", "-", "N", "E"}:
+                    parsed_data.append((term, op_label, index))
+                prev_index = index
+
+        self._register_length = max_index + 1
```

Comment:
I am not sure how I had missed this during the review of the `SpinOp` PR but I assume that this is what you refer to with the `I_1` problem above.
I think it ought to be necessary that the user can simply specify/overwrite the `register_length` himself when using sparse labels. Otherwise it is rather unintuitive that you need to add `I_{register_length - 1}` in order to encode the register length into a sparse label.
If we do not want to add a constructor argument then we need to document this more properly.